src/k4a_ros_device.cpp

```diff
@@ -281,6 +283,22 @@ K4AROSDevice::K4AROSDevice()
     body_index_map_publisher_ = image_transport::create_publisher(this,"body_index_map/image_raw");
   }
 #endif
+  
+  k4a_result_t result = this->startCameras();
+
+  if (result != K4A_RESULT_SUCCEEDED)
+  {
+    RCLCPP_ERROR_STREAM(this->get_logger(),"Failed to start cameras");
+  }
+
+  result = this->startImu();
+  if (result != K4A_RESULT_SUCCEEDED)
+  {
+    RCLCPP_ERROR_STREAM(this->get_logger(),"Failed to start IMU");
+  }
+
+  RCLCPP_INFO(this->get_logger(),"K4A Started");
```

Comment:
this logic was in the `k4a_ros_bridge_node` , which manually created the executor in some main function. This is not supported in composable nodes. So I moved everything here and stopped building that node. In general, the new way should be fine also for launching in non-composable way.